go/worker/common/committee/runtime_host.go

```diff
@@ -125,6 +106,99 @@ func (n *Node) GetRuntime() runtimeRegistry.Runtime {
 	return n.Runtime
 }
 
+// computeRuntimeHostNotifier is a runtime host notifier suitable for compute
+// runtimes.
+type computeRuntimeHostNotifier struct {
+	ctx context.Context
+
+	logger *logging.Logger
+
+	runtime    runtimeRegistry.Runtime
+	host       host.Runtime
+	keyManager keymanagerApi.Backend
+}
+
+func (h *computeRuntimeHostNotifier) watchPolicyUpdates() {
+	// Wait for the runtime.
+	rt, err := h.runtime.RegistryDescriptor(h.ctx)
+	if err != nil {
+		h.logger.Error("failed to wait for registry descriptor",
+			"err", err,
+		)
+		return
+	}
+	if rt.KeyManager == nil {
+		h.logger.Info("no keymanager needed, not watching for policy updates")
+		return
+	}
+
+	stCh, stSub := h.keyManager.WatchStatuses()
+	defer stSub.Close()
+	h.logger.Info("watching policy updates", "keymanager_runtime", rt.KeyManager)
+
+	for {
+		select {
+		case <-h.ctx.Done():
+			return
+		case st := <-stCh:
+			h.logger.Info("got policy update", "status", st)
+
+			// Ignore status updates if key manager is not yet known (is nil)
+			// or if the status update is for a different key manager.
+			if !st.ID.Equal(rt.KeyManager) {
+				continue
+			}
+
+			raw := cbor.Marshal(st.Policy)
+			req := &protocol.Body{RuntimeKeyManagerPolicyUpdateRequest: &protocol.RuntimeKeyManagerPolicyUpdateRequest{
+				SignedPolicyRaw: raw,
+			}}
+
+			var response *protocol.Body
+			call := func() error {
+				resp, err := h.host.Call(h.ctx, req)
+				if err != nil {
+					h.logger.Error("failed to dispatch RPC call to runtime",
+						"err", err,
+					)
+					return err
+				}
+				response = resp
+				return nil
+			}
+
+			retry := backoff.WithMaxRetries(backoff.NewConstantBackOff(retryInterval), maxRetries)
+			err := backoff.Retry(call, backoff.WithContext(retry, h.ctx))
+			if err != nil {
+				h.logger.Error("backoff error",
```

Comment:
Probably should say something like "failed dispatching key manager policy update to runtime".